### PR TITLE
Reusable EventList component

### LINKED ISSUE #2594: Reusable EventList component
To compose UI interactions with the search result event list we need to refactor it to its own, reusable component.

The high level design idea is to encapsulate much of the search logic within the component so that the parent only need to supply a query string, filter or DSL.

### PR DESCRIPTION
**Pull request by berggren**
This PR refactors the EventList component. It used to be embedded in the Explore view. This limits its use and featrues such as Graphs and Stories cannot use it.

closes #2594

### CODE REVIEW

File: timesketch/frontend-ng/src/views/Sketch.vue [hunk 1]
```diff
@@ -291,7 +298,7 @@ export default {
     return {
       showSketchMetadata: false,
       navigationDrawer: {
-        width: 400,
+        width: 370,
```
Comment (jkppr) [inline timesketch/frontend-ng/src/views/Sketch.vue:301]: Changing the width of the left panel back to 370 will result in an odd line break in the left panel when the pagination is used. (The next page arrow gets moved to a new line, see screenshot below)
page

File: timesketch/frontend-ng/src/components/Explore/EventList.vue [hunk 2]
```diff
@@ -0,0 +1,946 @@
...
+          value: '_source.timestamp',
+          width: '200',
+        },
+        {
+          value: '_source.comment',
+          width: '40',
+        },
+      ]
+      let extraHeaders = []
+      this.selectedFields.forEach((field) => {
```
Comment (jkppr) [inline timesketch/frontend-ng/src/components/Explore/EventList.vue:518]: The changes introduced in the [last commit](https://github.com/google/timesketch/pull/2592/commits/4efc22847b2f54786dc731e59565917799cbe725) result in `this.selectedFields` being undefined at this point. 
This results in a broken search and TS throwing the following error:
```
Error in render: "TypeError: Cannot read properties of undefined (reading 'forEach')"
found in

---> <TsEventList> at src/components/Explore/EventList.vue
       <VCard>
         <Explore> at src/views/Explore.vue
           <Canvas> at src/views/Canvas.vue
             <Sketch> at src/views/Sketch.vue
               <VMain>
                 <VApp>
                   <App> at src/App.vue
                     <Root>
```
and
```
 TypeError: Cannot read properties of undefined (reading 'forEach')
    at VueComponent.headers (EventList.vue?1150:519:1)
```

**Review by jkppr — CHANGES_REQUESTED** [review]
The latest commit has introduced a breaking change. Please take a look.

**Review by jkppr — APPROVED** [review]
Just one small comment. 
Overall LGTM

File: timesketch/frontend-ng/src/views/Sketch.vue (hunk 1 above)
Comment (berggren) [reply timesketch/frontend-ng/src/views/Sketch.vue:301]: Ah, makes sense. Change it to 400 as default

**Pull request merged**